rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -1095,80 +1095,30 @@ bool main_window::HandlePackageInstallation(QStringList file_paths, bool from_bo
 			bootable_paths_installed[bootable_paths[index]] = packages[index].title_id;
 		}
 
-		pdlg.hide();
+		const bool installed_a_whole_package_without_new_software = bootable_paths_installed.empty() && !cancelled;
 
-		if (!cancelled || !bootable_paths_installed.empty())
+		if (!bootable_paths_installed.empty())
 		{
-			if (bootable_paths_installed.empty())
+			m_game_list_frame->AddRefreshedSlot([paths = std::move(bootable_paths_installed)]() mutable
 			{
-				m_gui_settings->ShowInfoBox(tr("Success!"), tr("Successfully installed software from package(s)!"), gui::ib_pkg_success, this);
-				return true;
-			}
-
-			auto dlg = new QDialog(this);
-			dlg->setWindowTitle(tr("Success!"));
-
-			QVBoxLayout* vlayout = new QVBoxLayout(dlg);
-
-			QCheckBox* desk_check = new QCheckBox(tr("Add desktop shortcut(s)"));
-#ifdef _WIN32
-			QCheckBox* quick_check = new QCheckBox(tr("Add Start menu shortcut(s)"));
-#elif defined(__APPLE__)
-			QCheckBox* quick_check = new QCheckBox(tr("Add dock shortcut(s)"));
-#else
-			QCheckBox* quick_check = new QCheckBox(tr("Add launcher shortcut(s)"));
-#endif
-			QLabel* label = new QLabel(tr("Successfully installed software from package(s)!\nWould you like to install shortcuts to the installed software? (%1 new software detected)\n\n").arg(bootable_paths_installed.size()), dlg);
-
-			vlayout->addWidget(label);
-			vlayout->addStretch(10);
-			vlayout->addWidget(desk_check);
-			vlayout->addStretch(3);
-			vlayout->addWidget(quick_check);
-			vlayout->addStretch(3);
-
-			QDialogButtonBox* btn_box = new QDialogButtonBox(QDialogButtonBox::Ok);
-
-			vlayout->addWidget(btn_box);
-			dlg->setLayout(vlayout);
-
-			bool create_desktop_shortcuts = false;
-			bool create_app_shortcut = false;
-
-			connect(btn_box, &QDialogButtonBox::accepted, this, [&]()
-			{
-				create_desktop_shortcuts = desk_check->isChecked();
-				create_app_shortcut = quick_check->isChecked();
-				dlg->accept();
+				ShowOptionalGamePreparations(tr("Success!"), tr("Successfully installed software from package(s)!"), std::move(paths));
 			});
+		}
 
-			dlg->setAttribute(Qt::WA_DeleteOnClose);
-			dlg->exec();
+		m_game_list_frame->Refresh(true);
 
-			std::set<gui::utils::shortcut_location> locations;
-#ifdef _WIN32
-			locations.insert(gui::utils::shortcut_location::rpcs3_shortcuts);
-#endif
-			if (create_desktop_shortcuts)
-			{
-				locations.insert(gui::utils::shortcut_location::desktop);
-			}
-			if (create_app_shortcut)
-			{
-				locations.insert(gui::utils::shortcut_location::applications);
-			}
+		std::this_thread::sleep_for(std::chrono::microseconds(100'000 - std::min<usz>(100'000, get_system_time() - start_time)));
```

Comment:
What is this for again?
Please add a comment.

Reply:
I think it's so you had the 100% state for a little while, but this was in addition to the full refresh afterwards so it wasn't very efficient